### PR TITLE
[DOCS] Reformat `pattern_replace` token filter

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:

* Rewrites description and adds Lucene link
* Adds analyze example
* Adds parameter definitions
* Adds custom analyzer example

### CODE REVIEW

File: docs/reference/analysis/tokenfilters/pattern_replace-tokenfilter.asciidoc [hunk 1]
```diff
@@ -4,23 +4,157 @@
...
-StackOverflowError and cause the node it is running on to exit suddenly.
+The following <<indices-create-index,create index API>> request
+configures a new <<analysis-custom-analyzer,custom analyzer>> using a custom
+`pattern_replace` filter, `my_pattern_replace_filter`.
 
-Read more about http://www.regular-expressions.info/catastrophic.html[pathological regular expressions and how to avoid them].
+The `my_pattern_replace_filter` filter uses the regular expression `\\p{Sc}` to
+match and remove currency symbols, such as `£`, `€`, or `$`. The filter's `all`
+parameter is `false`, meaning only the first matching symbol in each token is
+removed.
```
Comment (markharwood) [inline docs/reference/analysis/tokenfilters/pattern_replace-tokenfilter.asciidoc:133]: I did not know `\\p{Sc}` was a thing so thanks for educating me :) 
While these more esoteric functions work in token filters they are not supported in [RegExp](https://www.elastic.co/guide/en/elasticsearch/reference/current/regexp-syntax.html) query syntax.

I wonder if it's worth doing one of two things here:
A) Add a footnote to the effect of "* note this particular expression wouldn't work with RegExp queries which use a different regex parser" or
B) Use a different example that works in both "pattern_replace" token filters and RegExp queries.

Comment (jrodewig) [reply docs/reference/analysis/tokenfilters/pattern_replace-tokenfilter.asciidoc:133]: Thanks @markharwood. With 992ab1cd8e4, I updated the example to use a pattern that compatible with the RegExp query.

**Review by markharwood — APPROVED** [review]
LGTM.
Thanks, @jrodewig

**Comment by jrodewig** [comment]
Thanks @markharwood.

**Comment by jrodewig** [comment]
### Backport commits
master bb66d59
7.x  c36df27 
7.8 3255581
7.7 b41f285

**Pull request merged**